allow installing non-version locked package on Red Hat with repo method

lukebigum opened the pull request:
Can install a package from a repo that doens't follow the $version-$release format.

Also locked json_pure to compatible version for rbuy 1.9 tests.

bastelfreak committed:
add initial archlinux support

adamcstephens commented:
You pulled in some other commits with this PR.

This is the best way submitted so far for solving this. #60 and #76 both don't account for version-iteration so should be closed.

dhoppe committed:
Update based on voxpupuli/modulesync_config 0.16.3

dhoppe committed:
Fix several Lint issues

dhoppe committed:
Fix several Rubocop issues

dhoppe committed:
Fix open ended dependency for puppet/archive

dhoppe committed:
Rename directory tests to examples

dhoppe committed:
Fix several Markdown issues

dhoppe committed:
Fix namespace

dhoppe committed:
Add support for rspec-puppet-facts

dhoppe committed:
Fix issue regarding non numeric versions

dhoppe committed:
Add support for anchor pattern

bastelfreak committed:
Merge pull request #1 from dhoppe/modulesync

Fix open ended dependency for puppet/archive

andyroyle committed:
fix function calls to send_request in datasource provider

bastelfreak committed:
modulesync 0.16.4

alexjfisher committed:
Merge pull request #3 from voxpupuli/modulesync

modulesync 0.16.4

bastelfreak committed:
modulesync 0.16.6

bbriggs committed:
Merge pull request #4 from voxpupuli/modulesync

modulesync 0.16.6

juniorsysadmin committed:
Bump min version_requirement for Puppet + deps

We currently only run automated tests against Puppet 3 latest and
therefore cannot guarantee that this module works as is expected with
earlier Puppet versions

Bump dependencies to the minimum version that should work under
Puppet 4, based on the metadata

Also remove deprecated pe version_requirement field

bastelfreak committed:
Merge pull request #5 from juniorsysadmin/bumpdeps

Bump min version_requirement for Puppet + deps

andyroyle committed:
fix missing parameter on send_request method call

andyroyle committed:
name methods to match property 'is_default'

andyroyle committed:
disable the predicateName rule for grafana_datasource.rb

bastelfreak committed:
modulesync 0.16.7

alexjfisher committed:
Merge pull request #7 from voxpupuli/modulesync

modulesync 0.16.7

andyroyle committed:
only suppress the predicateName rule for the given methods

doomnuggets committed:
Update config.ini.erb

bastelfreak committed:
modulesync 0.16.8

nibalizer committed:
Merge pull request #13 from voxpupuli/modulesync

modulesync 0.16.8

nibalizer committed:
Merge pull request #11 from doomnuggets/master

Update config.ini.erb

nibalizer committed:
Merge pull request #6 from andyroyle/fix-datasource-provider

fix datasource provider

igalic committed:
Add changelog entries for changes since 2.5.0

this includes our changes since the fork

bastelfreak committed:
Merge pull request #14 from igalic/changelog

Add changelog entries for changes since 2.5.0

goya151 committed:
Update metadata.json

Release 2.6.0

igalic committed:
Merge pull request #16 from goya151/master

Update metadata.json

igalic committed:
[blacksmith] Bump version to 2.6.1-rc0

Commit:
Add grafana_api_path to allow for API sub-paths

- Updated types and providers to allow the grafana_api_path parameter to
be set when using sub-paths for the API

bastelfreak committed:
release 2.6.1

hunner committed:
Merge pull request #18 from voxpupuli/rel261

release 2.6.1

bastelfreak committed:
bump required puppet version to 4.6.1

bastelfreak committed:
modulesync 0.19.3

dhollinger committed:
Merge pull request #19 from voxpupuli/modulesync

Modulesync 0.19.3

rnelson0 committed:
[blacksmith] Bump version to 2.6.2-rc0

rnelson0 committed:
Revert "[blacksmith] Bump version to 2.6.2-rc0"

This reverts commit 256f723029b06d9372d7e8eea88216f98b58d84b.

rnelson0 committed:
[blacksmith] Bump version to 2.6.2-rc0

rnelson0 committed:
The puppet 4-only release will start at 3.0.0

bastelfreak committed:
Merge pull request #21 from voxpupuli/3.0.0-rc0

The puppet 4-only release will start at 3.0.0

Commit:
Use operatingsystemmajrelease fact in el repo url

bastelfreak committed:
Merge pull request #24 from mirekys/master

Use operatinsystemmajrelease fact in repo url

bastelfreak committed:
modulesync 0.20.0

bastelfreak committed:
delete unneeded spec.opts file

sacres committed:
Merge pull request #26 from voxpupuli/modulesync

modulesync 0.20.0

Commit:
supoort jessie for install method repo

Commit:
FIX configuration file ownership

Depending on the default ownership and permissions of the Puppet run, the
configuration might be unreadable by the grafana user, causing the service
startup to fail.

One such possible cause is if the file type has default permissions such as:

File {
  owner => 'root',
  group => 'root',
  mode  => '0600',
}

bastelfreak committed:
Merge pull request #30 from hmrc/fix_config_file_ownership

FIX configuration file ownership

Commit:
FIX grafana_dashboards crash

This commit fixes the crash on the to_s method.

It also fixes the API endpoint for dashboards with + character in the name

Commit:
FIX API requests

Commit:
FIX idempotency of grafana_dashboard

roidelapluie committed:
Merge pull request #31 from hmrc/fix_dashboards

Fix grafana_dashboards

bastelfreak committed:
enhance spec tests

bbriggs committed:
Merge pull request #32 from bastelfreak/addarchsupport

Add support for archlinux

bastelfreak committed:
modulesync 0.20.1

bastelfreak committed:
fix package ensure for Archlinux

we can't pass the grafana version to the archlinux package provider. it
has no versionable feature. this adds a workaround to provide 'present'
to it.

vinzent committed:
Merge pull request #33 from voxpupuli/modulesync

modulesync 0.20.1

bastelfreak committed:
increase spec coverage

alexjfisher committed:
Merge pull request #34 from bastelfreak/fix_archlinux

implement package_ensure param for archlinux

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge branch 'master' into debian-repo-fix

Commit:
update spec definitions for debian apt repo

dhoppe committed:
Merge pull request #28 from roock/debian-repo-fix

supoort jessie for install method repo

dhoppe committed:
Debian and RedHat based operating systems should use the repository by default

bastelfreak committed:
Merge pull request #36 from dhoppe/repo

Debian and RedHat based operating systems should use the repository by default

dhoppe committed:
Bump version, Update changelog

dhoppe committed:
Fix serveral lint issues

bastelfreak committed:
Merge pull request #38 from dhoppe/release

Bump version, Update changelog

furhouse committed:
Always use jessie apt repo, even if even if you are on Ubuntu or another Debian version, according to docs.grafana.org/installation/debian.

juniorsysadmin committed:
Merge pull request #41 from furhouse/feature/ubuntu

Always use jessie apt repo when osfamily is Debian.

bastelfreak committed:
modulesync 0.21.0

bastelfreak committed:
rubocop: autofixes

dhoppe committed:
Merge pull request #42 from voxpupuli/modulesync

Modulesync 0.21.0

bastelfreak committed:
drop newvalues for type property in datasource

We want to support custom plugins, they may add new datasources, so a
hardcoded list is a bad idea.

bastelfreak committed:
Add defined resource for custom plugins

This is a simple wrapper around the grafana CLI. It allows us to install
custom plugins.

bastelfreak committed:
update README.md for grafana::plugin

bastelfreak committed:
add tests for grafana:plugin

bastelfreak committed:
Merge pull request #44 from bastelfreak/support_custom_plugins

Support custom plugins

bastelfreak committed:
modulesync 0.21.3

bastelfreak committed:
disable rubocop false/positive

bbriggs committed:
Merge pull request #46 from voxpupuli/modulesync

modulesync 0.21.3

bastelfreak committed:
release 4.0.0

ghoneycutt committed:
Support newer versions of puppetlabs/apt module

wyardley committed:
Merge pull request #53 from ghoneycutt/support_newer_apt

Support newer versions of puppetlabs/apt module

Commit:
modulesync 2017-09-15

wyardley committed:
Merge pull request #55 from voxpupuli/modulesync

modulesync 1.1.0

atward committed:
Fix typo in provider

wyardley committed:
Merge pull request #58 from atward/provider-fix

Fix typo in provider

Commit:
Create grafana_plugin resource type, and create grafana::plugins passthrough

Commit:
Bump version to 4.5.1.

Commit:
Add acceptance tests to run in Travis.

Commit:
Rework how classes are ordered

atward committed:
grafana_user custom resource

Commit:
updates to metadata

bastelfreak committed:
Merge pull request #61 from wyardley/default_package_version

BREAKING: Update default Grafana version to 4.5.1 and improve acceptance tests

Commit:
Remove licenses from the top of files, since LICENSE is already included in the project

wyardley committed:
Merge pull request #60 from atward/provider-user

grafana_user custom resource

Commit:
Get rid of the dependency on 'wget' module since archive is already a dependency

bastelfreak committed:
Merge pull request #63 from wyardley/plugin_types

BREAKING: Create grafana_plugin resource type and change grafana::plugins

bastelfreak committed:
Merge pull request #65 from wyardley/remove_wget

Get rid of the dependency on 'wget' module in favor of puppet-archive

Commit:
Switch to Puppet Data Types

bastelfreak committed:
Merge pull request #66 from wyardley/data_types

BREAKING: Switch to Puppet Data Types (ldap_cfg is now undef when disabled)

Commit:
Update README.md

bastelfreak committed:
Merge pull request #64 from wyardley/remove_license

Remove licenses from the top of files

bastelfreak committed:
Merge pull request #67 from wyardley/update_readme

Update README.md

wyardley committed:
Merge pull request #62 from bastelfreak/rel400

Release 4.0.0

bastelfreak committed:
[blacksmith] Bump version to 4.0.1-rc0

Commit:
Update Travis secrets

Commit:
CHANGELOG and metadata for release

alexjfisher committed:
Merge pull request #68 from wyardley/release401

Release 4.0.1

Commit:
[blacksmith] Bump version to 4.0.2-rc0

Commit:
fix typo in metadata (redhat 6 twice vs 6/7)

TraGicCode committed:
Merge pull request #69 from wyardley/fix_metadata_typo

fix typo in metadata (redhat 6 twice vs 6/7)

bastelfreak committed:
add debian 8 and 9 support

bastelfreak committed:
bump archive upper boundary to work witha latest versions

dhollinger committed:
Merge pull request #72 from bastelfreak/debian

add debian 8 and 9 support

dhollinger committed:
Merge pull request #73 from bastelfreak/archive

bump archive upper boundary to work with latest versions

Commit:
Release 4.0.2

llowder committed:
Merge pull request #74 from dhollinger/rel402

Release 4.0.2

bastelfreak committed:
[blacksmith] Bump version to 4.0.3-rc0

Commit:
Removing the datasource url check as it leads to errors with postgres-datasource

dhollinger committed:
Merge pull request #76 from Faffnir/master

Removing the datasource url check as it leads to errors with postgres…

bastelfreak committed:
modulesync 1.5.0

bastelfreak committed:
Merge pull request #78 from voxpupuli/modulesync

modulesync 1.5.0

devcfgc committed:
update readme with examples of using grafana_datasource and grafana_dashboard

bastelfreak committed:
release 4.0.3

devcfgc committed:
fixing puppet-lint

bastelfreak committed:
Merge pull request #80 from devcfgc/update_readme

Update readme with examples of using datasource and dashboard

Commit:
(SIMP-4206) Added Organization Provider and updated Datasource Provider

- Updated grafana datasource provider to allow datasources to be created
on specific organizations
- Added grafana organization provider to allow organizations to be
created
- Updated README to add more descriptive information for previous
features as well as new

Fixes commercial/enterprise-meta#91
SIMP-4206 #close

bastelfreak committed:
Merge pull request #81 from bastelfreak/rel410

release 4.0.3

bastelfreak committed:
[blacksmith] Bump version to 4.0.4-rc0

bastelfreak committed:
modulesync 1.6.0

bastelfreak committed:
Merge pull request #84 from voxpupuli/modulesync

modulesync 1.6.0

Commit:
Updating to fix spec tests/rubocop errors

Commit:
final rubocop fix

bastelfreak committed:
Merge pull request #86 from simp/SIMP-4206

(SIMP-4206) Added organization provider and updated datasource provider

bastelfreak committed:
modulesync 1.7.0

bastelfreak committed:
Merge pull request #87 from bastelfreak/modulesync

modulesync 1.7.0

bastelfreak committed:
release 4.1.0

bastelfreak committed:
Merge pull request #88 from bastelfreak/rel410

release 4.1.0

bastelfreak committed:
[blacksmith] Bump version to 4.1.1-rc0

Commit:
fix datasource provider error

with_credentials def was referencing datasource[:is_default] and
preventing idempotency

bastelfreak committed:
Merge pull request #90 from brandonrdn/master

fix datasource provider error

bastelfreak committed:
modulesync 1.8.0

bastelfreak committed:
release 4.1.1

bastelfreak committed:
Merge pull request #91 from voxpupuli/modulesync

modulesync 1.8.0

bastelfreak committed:
Merge pull request #92 from voxpupuli/rel401

release 4.1.1

bastelfreak committed:
[blacksmith] Bump version to 4.1.2-rc0

bastelfreak committed:
Merge pull request #93 from brandonrdn/master

Add grafana_api_path to allow for API sub-paths

Commit:
Expand organization property for dashboards

Commit:
Bug fixes and adding organization option to dashboards

bastelfreak committed:
Merge pull request #94 from brandonrdn/master

Expand organization property for dashboards

bastelfreak committed:
release 4.2.0

bastelfreak committed:
Merge pull request #97 from bastelfreak/rel420

release 4.2.0

bastelfreak committed:
[blacksmith] Bump version to 4.2.1-rc0

cclloyd committed:
Update README.md

Fixed typos on all lines that defined grafana_api_path

juniorsysadmin committed:
Merge pull request #99 from cclloyd/patch-1

Update README.md

bastelfreak committed:
switch from topscope facts to $facts hash

bastelfreak committed:
Merge pull request #102 from bastelfreak/facts

switch from topscope facts to $facts hash

TraGicCode committed:
modulesync 1.9.0

Faffnir committed:
Adding Postgres support

Faffnir committed:
Adding secure_json_data field to the grafana_datasource declaration

Faffnir committed:
Removing this check again as it breaks postgres datasource definition

Removed this explicitly in commit d7e19910cfad56fd87c9c5047c621646f5105bf9 but the check was readded...

bastelfreak committed:
Merge pull request #105 from Faffnir/master

Add postgres support and secure_json_data support

bastelfreak committed:
modulesync 1.9.0-12-gacb0134

bastelfreak committed:
Merge pull request #106 from voxpupuli/modulesync

modulesync 1.9.1

ekohl committed:
Rely on beaker-hostgenerator for docker nodesets

bastelfreak committed:
Merge pull request #108 from ekohl/msync-docker

Rely on beaker-hostgenerator for docker nodesets

bastelfreak committed:
drop EOL OSs; fix puppet version range

bastelfreak committed:
Merge pull request #109 from bastelfreak/puppet

drop EOL OSs; fix puppet version range

Commit:
Updated release codename from jessie to stretch.

Grafana upstream documetation: http://docs.grafana.org/installation/debian/, says to use 'stretch' codename for all repositories.

bastelfreak committed:
Merge pull request #113 from drshawnkwang/fix_debian-update-release-codename

Update release codename from jessie to stretch.

bastelfreak committed:
release 4.3.0

sebastianrakel committed:
Merge pull request #114 from bastelfreak/rel430

release 4.3.0

bastelfreak committed:
[blacksmith] Bump version to 4.3.1-rc0

ZeroPointEnergy committed:
Manage sysconfig files

Use augeas to manage the sysconfig files which come with the
Debian and RedHat packages.

bastelfreak committed:
Merge pull request #115 from ZeroPointEnergy/feature/sysconfig

Manage sysconfig files

Commit:
Puppet provisioning of daashboards and datasources.

Add funtionality to allow puppet to provision datasources and dashboards using grafana's built-in provisioning feature.
Add templates for datasources and dashbaords.
Allow dashboards to be dynamically provisioned using puppet file resource.

bastelfreak committed:
bump archive upper version boundary to <4.0.0

bastelfreak committed:
Merge pull request #116 from bastelfreak/archive

bump archive upper version boundary to <4.0.0

bastelfreak committed:
release 4.4.0

llowder committed:
Merge pull request #117 from bastelfreak/rel440

release 4.4.0

bastelfreak committed:
[blacksmith] Bump version to 4.4.1-rc0

Commit:
Added a file resource to config function.
Modified dependency slightly to allow plugins to be installed after file resource creates plugin directory: only works for package, repo, and archive install methods.

Commit:
Add Ruby function and puppet code to extract path from hash.

Ruby function copies the options array from the input hash for puppet. Puppet uses this to create file resource for dashboard directories. Finally, the hash without the 'puppetsource' key is used in the template for grafana.

Commit:
Documentation on provisioning. No functional changes.

Commit:
Fixed whitespace issue in resources.

bastelfreak committed:
Merge pull request #118 from drshawnkwang/fix_bug-dependencies

Fix dependency in provisioning plugins

bastelfreak committed:
release 4.4.1

bastelfreak committed:
Merge pull request #119 from bastelfreak/rel441

release 4.4.1

bastelfreak committed:
[blacksmith] Bump version to 4.4.2-rc0

Commit:
Added functionality to create all subpaths given in options path variable. Only works on UNIX paths.
Added parameter to control whether the creation of the subpaths is to be controlled by puppet.

Commit:
Fixed errors found by Travis CI testing.

Commit:
Additional changes for Travis CI - ruby style/syntax

Commit:
metadata.json - bumped version requirement for puppetlabs-stdlib
Changed spacing in params.pp
Removed absolute top-scope and changed templates to use epp files
Refactored .erb templates into .epp templates.

Commit:
Changed spacing to resolve Travis CI build error.

Commit:
Added missing comma to docs. No functional changes.

Commit:
Added acceptance test, written by @bastelfreak.

bastelfreak committed:
Merge pull request #120 from drshawnkwang/feature_useprovision

Feature: Add grafana provisioning to this module.

bastelfreak committed:
modulesync 1.9.3

bastelfreak committed:
Merge pull request #123 from voxpupuli/modulesync

modulesync 1.9.3

bastelfreak committed:
release 4.5.0

bastelfreak committed:
Merge pull request #124 from bastelfreak/rel450

release 4.5.0

bastelfreak committed:
[blacksmith] Bump version to 4.5.1-rc0

baurmatt committed:
Change default of version parameter to 'installed'

Fixes GH-125.

kazeborja committed:
Parametrize provisioning files names

kazeborja committed:
Update documentation about provisioning paths

kazeborja committed:
Add acceptance test for provisioning paths

bastelfreak committed:
modulesync 1.9.4

bastelfreak committed:
Merge pull request #129 from voxpupuli/modulesync

modulesync 1.9.4

bastelfreak committed:
Merge pull request #128 from kazeborja/parametrizeProvisioning

Parametrize provisioning file names

bastelfreak committed:
modulesync 1.9.6

bastelfreak committed:
update spec_helper_acceptance

bastelfreak committed:
Merge pull request #131 from voxpupuli/modulesync

modulesync 1.9.6

rwuest committed:
added optional --repo option to grafana_cli plugin install

bastelfreak committed:
Merge pull request #132 from rwuest/plugin_repo_param

add --repo option to grafana_cli plugin install

bastelfreak committed:
allow puppetlabs/stdlib 5.x

bastelfreak committed:
allow puppetlabs/apt 6.x

Commit:
Merge pull request #134 from voxpupuli/stdlib

allow puppetlabs/stdlib 5.x and puppetlabs/apt 6.x

bastelfreak committed:
Merge pull request #126 from syseleven/fix-version-parameter

Change default of version parameter to 'installed'

bastelfreak committed:
modulesync 2.0.0

bastelfreak committed:
Merge pull request #135 from voxpupuli/modulesync

modulesync 2.0.0
